Robust_Lab.qmd

```diff
@@ -12,93 +13,368 @@ execute:
+
 ## Bootstrapping and Permutations
 
 2.  For the following data: \[8.453532, 10.025041, 11.495339, 9.367600, 8.333229, 9.788753, 10.883344, 10.543059, 9.869095, 10.799819\]
 
     a\. Bootstrap the mean (using the `boot` package) and plot the histogram with `ggplot2`
 
+    ```{r}
+    # create list
+    sample_list <- c(8.453532, 10.025041, 11.495339, 9.367600, 8.333229, 9.788753, 10.883344, 10.543059, 9.869095, 10.799819)
+
+    # set seed
+    set.seed(8)
+
+    # method 1: Using `simpleboot` package
+    simpleboot_list <- one.boot(sample_list, mean, R=1000)
+    hist(simpleboot_list)
+
+    # method 2: Using `boot` package
+    # define the function that will be used to calculate the mean
+    mean_func = function(data, indices) {
+      return(mean(data[indices])) #indices to do bootstrapping
+    }
+    # use boot to bootstrap means 1000 times
+    boot_list_results <- boot(sample_list, mean_func, R=1000)
+    # get means
+    boot_list_means <- boot_list_results$t
+
+    # plot histogram with ggplot
+    p <- ggplot(as.data.frame(boot_list_means), aes(x = V1)) + 
+      geom_histogram(binwidth = 0.1, 
+                     color = "black", 
+                     fill = "#1E88E5", # color-blind friendly
+                     alpha = 0.4) +
+      labs(title="Figure 2: Distribution of bootstrapped means",
+           y="Frequency",
+           x="Bootstrap Sample Means") +
+      theme_classic()
+
+    # add vline
+    vline <- mean(sample_list)
+    p + geom_vline(xintercept = vline,
+                   color = "#D81B60") + # color-blind friendly
+      annotate("text", x=vline-0.05, y=50,
+               color = "#D81B60",
+               label="Original sample mean", angle=90)
+    ```
+
     b\. Bootstrap the median (using the `boot` package) and plot the histogram with `ggplot2`
 
+    ```{r}
+
+    # define function to calculate the median
+    median_func = function(data, indices) {
+      return(median(data[indices])) #indices to do bootstrapping
+    }
+    # set seed
+    set.seed(8)
+    # use boot to bootstrap medians 1000 times
+    boot_list_median_results <- boot(sample_list, median_func, R=1000)
+    # get medians
+    boot_list_medians <- boot_list_median_results$t
+
+    # plot histogram with ggplot
+    p <- ggplot(as.data.frame(boot_list_medians), aes(x = V1)) + 
+      geom_histogram(binwidth = 0.1, 
+                     color = "black", 
+                     fill = "#D81B60", # color-blind friendly
+                     alpha = 0.4) +
+      labs(title="Figure 3: Distribution of bootstrapped medians",
+           y="Frequency",
+           x="Bootstrap Sample Medians") +
+      theme_classic()
+
+    # add vline
+    vline <- median(sample_list)
+    p + geom_vline(xintercept = vline,
+                   color = "#D81B60") + 
+      annotate("text", x=vline-0.05, y=100,
+               color = "black",
+               label="Original sample median", angle=90)
```

Comment:
I like the text to be horizontal. I think it looks better. If you remove the angle argument I think it does that.